freetextresponse/private/view.js

```diff
@@ -15,6 +15,16 @@ function FreeTextResponseView(runtime, element) {
     var url = runtime.handlerUrl(element, 'submit');
     var urlSave = runtime.handlerUrl(element, 'save_reponse');
 
+    var xblockId = $element.attr('data-usage-id');
+    var cachedAnswerId = xblockId + '_cached_answer';
+    var problemProgressId = xblockId + '_problem_progress';
+    var usedAttemptsFeedbackId = xblockId + '_used_attempts_feedback';
+    if ($('body').data(cachedAnswerId) !== undefined) {
```

Comment:
I don't love having all of the values hooked onto the `<body>` element on the page. Since we have access to `$element` let's bind the values to that, so each xblock has it's own data attached...
ie:

```javascript
if ($element.data(cachedAnswerId) !== undefined) {
    // Your code here...
}
```